cmd/kubeadm/app/cmd/init_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+	"k8s.io/kubernetes/cmd/kubeadm/app/features"
+)
+
+const (
+	testInitConfig = `---
+apiVersion: kubeadm.k8s.io/v1beta1
+kind: InitConfiguration
+localAPIEndpoint:
+  advertiseAddress: "1.2.3.4"
+bootstrapTokens:
+- token: "abcdef.0123456789abcdef"
+nodeRegistration:
+  criSocket: /run/containerd/containerd.sock
+  name: someName
+---
+apiVersion: kubeadm.k8s.io/v1beta1
+kind: ClusterConfiguration
+controlPlaneEndpoint: "3.4.5.6"
+`
+)
+
+func TestNewInitData(t *testing.T) {
+	// create temp directory
+	tmpDir, err := ioutil.TempDir("", "kubeadm-init-test")
+	if err != nil {
+		t.Errorf("Unable to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// create config file
+	configFilePath := filepath.Join(tmpDir, "test-config-file")
+	cfgFile, err := os.Create(configFilePath)
+	if err != nil {
+		t.Errorf("Unable to create file %q: %v", configFilePath, err)
+	}
+	defer cfgFile.Close()
+	if _, err = cfgFile.WriteString(testInitConfig); err != nil {
+		t.Fatalf("Unable to write file %q: %v", configFilePath, err)
+	}
+
+	testCases := []struct {
+		name        string
+		args        []string
+		flags       map[string]string
+		validate    func(*testing.T, *initData)
+		expectError bool
+	}{
+		// Init data passed using flags
+		{
+			name: "pass without any flag (use defaults)",
+		},
+		{
+			name: "fail if unknown feature gates flag are passed",
+			flags: map[string]string{
+				options.FeatureGatesString: "unknown=true",
+			},
+			expectError: true,
+		},
+		{
+			name: "fail if deprecetes feature gates are set",
+			flags: map[string]string{
+				options.FeatureGatesString: fmt.Sprintf("%s=true", features.CoreDNS),
+			},
+			expectError: true,
+		},
+		{
+			name: "fails if invalid preflight checks are provided",
+			flags: map[string]string{
+				options.IgnorePreflightErrors: "all,something-else",
+			},
+			expectError: true,
+		},
```

Comment:
We should add a positive test here, that validates, that valid values for the `IgnorePreflightErrors` are observed. Can be just a single value, the idea is to test if the flag gets propagated and not dismissed.